File: app/src/main/java/lv/chi/giffoid/ui/mvp/BasePresenter.java
package lv.chi.giffoid.ui.mvp;

import io.reactivex.disposables.CompositeDisposable;

public abstract class BasePresenter<V extends MvpView> implements MvpPresenter<V> {
    protected CompositeDisposable compositeDisposable = new CompositeDisposable();
    private V view;

    @Override
    public void bind(final V view) {
        this.view = view;
    }

    @Override
    public void unbind() {
        compositeDisposable.clear();
        view = null;
    }

    public V view() {
        return view;
    }

}

File: app/src/main/java/lv/chi/giffoid/ui/mvp/BasePresenter.kt
package lv.chi.giffoid.ui.mvp

import android.support.annotation.CallSuper
import io.reactivex.disposables.CompositeDisposable

abstract class BasePresenter<V : MvpView> : MvpPresenter<V> {
    protected var compositeDisposable = CompositeDisposable()
    protected var view: V? = null

    @CallSuper
    override fun bind(view: V) {
        this.view = view
    }

    @CallSuper
    override fun unbind() {
        compositeDisposable.clear()
        this.view = null
    }
}
